Source/InterceptorStrategies.cs

```diff
@@ -102,7 +102,28 @@ internal class ExtractProxyCall : IInterceptStrategy
 
 		public InterceptionAction HandleIntercept(ICallContext invocation, InterceptorContext ctx, CurrentInterceptContext localctx)
 		{
-			localctx.Call = FluentMockContext.IsActive ? (IProxyCall)null : ctx.OrderedCalls.LastOrDefault(c => c.Matches(invocation));
+			if (FluentMockContext.IsActive)
+			{
+				localctx.Call = null;
+			}
+			else
+			{
+				IProxyCall lastMatchingSetup = null;
+				IProxyCall lastMatchingSetupForSameMethod = null;
+				foreach (var setup in ctx.OrderedCalls)
+				{
+					if (setup.Matches(invocation))
+					{
+						lastMatchingSetup = setup;
+						if (setup.Method == invocation.Method)
+						{
+							lastMatchingSetupForSameMethod = setup;
+						}
+					}
+				}
+				localctx.Call = lastMatchingSetupForSameMethod ?? lastMatchingSetup;
+			}
```

Comment:
At first glance, this looks much more involved than a simple `LastOrDefault`, but it has the advantage of only performing a single enumeration over `ctx.OrderedCalls` while looking for two different items.